test/test_base64.cpp

```diff
@@ -18,6 +20,14 @@ TEST_CASE("base64.infrastructure")
     detail::base64_test_tables();
     #endif
 }
+// C4_LITTLE_ENDIAN is not always defined in c4/cpu.hpp, so we use runtime
+// detection of endianness.
+bool is_le() {
+    unsigned long const v = 1UL;
+    unsigned char b[sizeof(v)];
+    std::memcpy(&b[0], &v, sizeof(v));
+    return !!b[0];
+}
```

Comment:
then maybe use this additionally to unclutter the tests below:
```c++
csubstr native(csubstr little_endian, csubstr big_endian)
{
    return is_le() ? little_endian : big_endian;
}
// ... and then
{1234, native("0gQ=", "BNI=")}
// instead of
{1234, csubstr(c4::detail::is_le() ? "0gQ=" : "BNI=")}
```

Reply:
Ok, that’s reasonable. I’ll push another commit implementing something like that.